dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -427,10 +428,13 @@ public VersionTypeValidation validateTestParameterSet(Set<SourceFile> sourceFile
             // Iterate over each call, determine dependencies
             Map<String, List<String>> callsToDependencies = wdlBridge.getCallsToDependencies(tempMainDescriptor.getAbsolutePath(), mainDescName);
             toolInfoMap = mapConverterToToolInfo(callsToDockerMap, callsToDependencies);
+
             // Get import files
             namespaceToPath = wdlBridge.getImportMap(tempMainDescriptor.getAbsolutePath(), mainDescName);
-        } catch (IOException | NoSuchElementException | WdlParser.SyntaxError e) {
-            throw new CustomWebApplicationException("could not process wdl into DAG: " + e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        } catch (WdlParser.SyntaxError e) {
+            throw new CustomWebApplicationException(this.WDL_PARSE_ERROR + e.getMessage(), HttpStatus.SC_BAD_REQUEST);
+        } catch (IOException | NoSuchElementException e) {
+            throw new CustomWebApplicationException("Could not process request, " + e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
Log the exception

Reply:
[Logged](https://github.com/dockstore/dockstore/blob/c5277c33e2a3bf271f3be4e33b742fb646b1a076/dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java#L437).